cpu/esp_common/freertos/task.c

```diff
@@ -131,7 +131,7 @@ TaskHandle_t xTaskGetCurrentTaskHandle(void)
 void vTaskDelay( const TickType_t xTicksToDelay )
 {
     DEBUG("%s xTicksToDelay=%d\n", __func__, xTicksToDelay);
-#if defined(MCU_ESP8266) && defined(MODULE_ESP_WIFI_ANY)
+#if defined(MODULE_ESP_WIFI_ANY)
     uint64_t us = xTicksToDelay * MHZ / xPortGetTickRateHz();
     xtimer_usleep(us);
 #endif
```

Comment:
The function is only used by the WiFi hardware driver. Probably we could enable it always and let the linker optimize out.

Reply:
Yea the less ifdefs the better.
If it's not used it will be thrown out anyway.

Reply:
Ah, now I remeber why it was embedded in `#ifdef ... #endif`. You get
```
undefined reference to `_xtimer_tsleep' 
esp_freertos_common.a(task.o): In function `vTaskDelay':
task.c:(.text.vTaskDelay+0x20): undefined reference to `_xtimer_tsleep'
```
because module `xtimer` is only pulled in when module `esp_wifi_any` is used. I would change it back to keep `xtimer` out if it is not really required.

Reply:
BTW, we should also have compilation tests for `esp_wifi` & Co, see PR #12971. Although in this case the problem occurred without ESP-specific modules the usual case is the opposite.